nomos-da/network/subnetworks-assignations/src/versions/v1.rs

```diff
@@ -0,0 +1,88 @@
+use crate::MembershipHandler;
+use libp2p_identity::PeerId;
+use std::collections::HashSet;
+
+/// Fill a `N` sized set of "subnetworks" from a list of peer ids members
+pub struct FillFromNodeList {
+    pub assignations: Vec<HashSet<PeerId>>,
+    pub subnetwork_size: usize,
+    pub replication_factor: usize,
+}
+
+impl FillFromNodeList {
+    pub fn new(peers: &[PeerId], subnetwork_size: usize, dispersal_factor: usize) -> Self {
+        Self {
+            assignations: Self::fill(peers, subnetwork_size, dispersal_factor),
+            subnetwork_size,
+            replication_factor: dispersal_factor,
+        }
+    }
+
+    fn fill(
+        peers: &[PeerId],
+        subnetwork_size: usize,
+        replication_factor: usize,
+    ) -> Vec<HashSet<PeerId>> {
+        assert!(!peers.is_empty());
```

Comment:
Nit: Because of HashSet (only one unique entry) the test bellow fails, because each subnet has only one entry when replication factor is 2.

Reply:
Edited comment, this happens when there's only one peer in the list.